### PR TITLE
Add github action for Snyk scanning

### PR DESCRIPTION
**Pull request by asjohnston-asf**
This doesn't implement any scheduled scanning, or reporting via the Snyk web interface, but it satisfies the need of preventing code with known high/critical vulnerabilities from being deployed.

Repository is configured with my personal SNYK_TOKEN; I hope we'll be able to provision a [service account](https://docs.snyk.io/features/user-and-group-management/managing-groups-and-organizations/service-accounts) for our Snyk organization, but I'm content to move forward as-is.

Reference for Snyk CLI commands: https://docs.snyk.io/snyk-cli/cli-reference

Reference for the .snyk policy file: https://docs.snyk.io/products/snyk-infrastructure-as-code/snyk-cli-for-infrastructure-as-code/iac-ignores-using-the-.snyk-policy-file

### CODE REVIEW

**Comment by jtherrmann** [comment]
> * we may want to push any findings/reports up to our Snyk organization?
I did a quick search for this in the Snyk docs but didn't find anything; do you already know how to do this?

> * we may want to scan on a schedule, as well as on push?
What would be the main advantages of scanning on a schedule? (I can think of a few possibilities.)

**Comment by asjohnston-asf** [comment]
Looks like there's no quick and easy way to publish the results of a `snyk test` CLI run to the web interface. We could set up a `snyk monitor`, but I now think that's above an beyond for the current PR.

Scanning on a schedule would let us catch any newly-published CVEs; otherwise we wouldn't detect them until a new commit is pushed. I'm content to move ahead without scanning on a schedule for now. Dependabot will already do this for our python dependencies, and we see automated `boto3` dependabot PRs every couple of days that would trigger the full snyk scanning suite.

**Comment by asjohnston-asf** [comment]
Regarding the high severity finding for our API Gateway authorizer, I have an open inquiry to ESDIS on how best to proceed, but I proposed adding `.snyk` [policy file](https://docs.snyk.io/products/snyk-infrastructure-as-code/snyk-cli-for-infrastructure-as-code/iac-ignores-using-the-.snyk-policy-file) with a rule to ignore this specific check.

**Comment by asjohnston-asf** [comment]
Regarding the high severity finding for our API Gateway authorizer, I have an open inquiry to ESDIS on how best to proceed, but I proposed adding `.snyk` [policy file](https://docs.snyk.io/products/snyk-infrastructure-as-code/snyk-cli-for-infrastructure-as-code/iac-ignores-using-the-.snyk-policy-file) with a rule to ignore this specific check.

File: .github/workflows/static-analysis.yml [hunk 1]
```diff
@@ -67,5 +67,24 @@ jobs:
           sed -i 's/"Resource": "${.*}"/"Resource": "foo:bar"/' apps/step-function.json
           statelint apps/step-function.json
 
+  snyk:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: snyk/actions/setup@master
```
Comment (asjohnston-asf) [inline .github/workflows/static-analysis.yml:74]: switch to pinned versions?

**Review by jacquelynsmale — COMMENTED** [review]
Just a period missing!

**Pull request merged**